js/geo/transform.js

```diff
@@ -434,5 +410,16 @@ Transform.prototype = {
         mat4.translate(m, m, [-this.x, -this.y, 0]);
 
         this.projMatrix = m;
+
+        // matrix for conversion from location to screen coordinates
+        var m2 = mat4.create();
+        mat4.scale(m2, m2, [this.width / 2, -this.height / 2, 1]);
+        mat4.translate(m2, m2, [1, -1, 0]);
+        this.pxMatrix = mat4.multiply(new Float64Array(16), m2, this.projMatrix);
+
+        // inverse matrix for conversion from screen coordinaes to location
+        var pxi = mat4.invert(new Float64Array(16), this.pxMatrix);
+        if (!pxi) throw new Error("failed to invert matrix");
+        this.pxMatrixInverse = pxi;
```

Comment:
[Do you think it'd be possible to refactor this code so that the comments aren't required](https://sourcemaking.com/refactoring/smells/comments)?